reactor-core/src/test/java/reactor/core/publisher/FluxReplayTest.java

```diff
@@ -512,4 +476,56 @@ public void scanSubscriber() {
         test.cancelled = true;
         Assertions.assertThat(test.scan(Scannable.Attr.CANCELLED)).isTrue();
     }
+
+	@Test
+	public void cacheSingleSubscriberWithMultipleRequestsDoesntHang() {
+		await().atMost(5, TimeUnit.SECONDS)
+		       .untilAsserted(() -> assertThat(
+				       Flux.range(0, 1000)
+				           .cache(5)
+				           .toStream(10)
+				           .collect(Collectors.toList()))
+				       .hasSize(1000)
```

Comment:
nit: took me a bit of time to realize that `.hasSize` is applied on `assertThat`'s result :D since the chain (`Flux.range(...).cache(...)...`) is rather long, maybe it would be better to extract it into variable?

Reply:
well, `toStream().collectList()` block and the goal of awaitility usage here is to ensure that the terminal operation on the converted stream doesn't hang. having it extracted in a variable would defeat the purpose... maybe a Timeout annotation on the test would be better here?

Reply:
you can extract it into a variable **inside** `untilAsserted` :)

Reply:
the more I look at it, the less I find awaitility a good fit for it... since it will try to re-execute the blocking call on polling... replacing with a `@Test(timeout=5000)`